lib/vector-core/src/event/metadata.rs

```diff
@@ -8,11 +8,22 @@ use shared::EventDataEq;
 /// and `struct LogEvent` types.
 #[derive(Clone, Debug, Default, Deserialize, PartialEq, PartialOrd, Serialize)]
 pub struct EventMetadata {
+    /// Used to store the datadog API from sources to sinks
+    #[serde(default, skip)]
+    pub datadog_api_key: Option<String>,
     #[serde(default, skip)]
     finalizers: EventFinalizers,
 }
 
 impl EventMetadata {
+    /// Build metadata with datadog api key only
+    pub fn new_with_datadog_api_key(api_key: String) -> Self {
```

Comment:
All the uses of this function appear to follow this pattern:
```rust
event.metadata_mut().merge(EventMetadata::new_with_datadog_api_key(something));
```
That goes to a lot of work to create, merge, and destroy a metadata structure just for the purpose of setting a field that can be updated directly. Is there another reason to prefer this pattern?

Reply:
Good catch, I write that before rebasing on top your changes about end-to-end acknowledgement, will rework accordingly.

Reply:
I went with a simplistic solution, it should be better.